src/app/app.component.ts

```diff
@@ -47,19 +65,29 @@ export class App {
     this.store$.dispatch(new LoadAppInfo());
   }
 
-  configureAccessibility() {
-    if (this.platform.is('ios') && window && window.MobileAccessibility) {
-      window.MobileAccessibility.updateTextZoom();
-      window.MobileAccessibility.getTextZoom(this.getTextZoomCallback);
-      window.MobileAccessibility.isDarkerSystemColorsEnabled(
-        (increasedContrast: boolean) => this.increasedContrast = increasedContrast);
-    }
-    if (typeof this.platform.resume.subscribe === 'function') {
-      this.platform.resume.subscribe(() => {
-        window.MobileAccessibility.usePreferredTextZoom(true);
-        window.MobileAccessibility.getTextZoom(this.getTextZoomCallback);
-      });
-    }
+  configurePlatformSubscriptions() {
+    const merged$ = merge(
+      this.platform.resume.pipe(map(this.onAppResumed)),
+      this.platform.pause.pipe(map(this.onAppSuspended)),
+    );
+    this.platformSubscription = merged$.subscribe();
+  }
+
+  onAppResumed = () => {
+    this.store$.dispatch(new AppResumed());
+    window.MobileAccessibility.usePreferredTextZoom(true);
+    window.MobileAccessibility.getTextZoom(this.getTextZoomCallback);
+  }
+
+  onAppSuspended = () => {
+    this.store$.dispatch(new AppSuspended());
+  }
+
+  configureAccessibility = () => {
+    window.MobileAccessibility.updateTextZoom();
```

Comment:
wondering about the removal of the window && window.MobileAccessibility line (51).
Is this because window is guaranteed to exists and this was an unnecessary check?

Reply:
Exactly. And we can assume `MobileAccessibility` will exist too unless the plugin fails.

Reply:
Also, we shouldn't really need to use `window.MobileAccessibility` as this should work:

```constructor(mobileAccessibility: MobileAccessibility) {
  this.platform.ready()
    .then(() => mobileAccessibility.getTextZoom())
    .then(e => this.zoomLevel = e);
}``` see more at https://github.com/ionic-team/ionic-native/issues/2138

...but it doesn't. The signature is wrong - it doesn't return a promise, it returns a callback ... but you can't pass it a function. Went down a rabbit hole trying to reference the plugin correctly. In the end, decided to leave the code as is because it works.

Reply:
How does this react in the browser , we know the plugin won't be there and that line will error